File: datasets/uploaded_project/datasets/vul.cs
// TestVul.cs
// Vulnerable example: SQL injection (string concat) and Process.Start with concatenated args

using System;
using System.Data.SqlClient;
using System.Diagnostics;
using System.Web;

public class TestVulHandler : IHttpHandler {
    public void ProcessRequest(HttpContext context) {
        var request = context.Request;
        var response = context.Response;

        string username = request.QueryString["username"]; // user-controlled
        string host = request.QueryString["host"]; // user-controlled

        // --- VULN: SQL Injection via string concatenation ---
        try {
            string connStr = "Server=(local);Database=Test;Trusted_Connection=True;";
            using (SqlConnection conn = new SqlConnection(connStr)) {
                conn.Open();
                string sql = "SELECT * FROM Users WHERE username = '" + username + "'";
                SqlCommand cmd = new SqlCommand(sql, conn);
                var reader = cmd.ExecuteReader();
                while (reader.Read()) {
                    response.Write("User: " + reader["username"]);
                }
            }
        } catch (Exception ex) {
            response.Write("DB error");
        }

        // --- VULN: Process.Start with concatenated args (command injection risk) ---
        try {
            // Dangerous: using user input directly in arguments can be abused
            ProcessStartInfo psi = new ProcessStartInfo("ping", "-n 1 " + host);
            psi.RedirectStandardOutput = true;
            psi.UseShellExecute = false;
            var process = Process.Start(psi);
            string output = process.StandardOutput.ReadToEnd();
            response.Write("<pre>" + HttpUtility.HtmlEncode(output) + "</pre>");
        } catch (Exception ex) {
            response.Write("Exec failed");
        }
    }

    public bool IsReusable { get { return false; } }
}

File: datasets/uploaded_project/datasets/vul.java

// test_vul.java
// Vulnerable example: SQL injection and Runtime.exec with user input

import java.sql.*;
import java.io.*;
import javax.servlet.*;
import javax.servlet.http.*;

public class VulnerableServlet extends HttpServlet {
    private Connection getConnection() throws SQLException {
        // Example connection. Replace with your DB config in real tests.
        return DriverManager.getConnection("jdbc:sqlite::memory:");
    }

    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        String user = req.getParameter("user"); // user-controlled

        // --- VULN: SQL Injection via string concatenation ---
        try (Connection conn = getConnection();
             Statement stmt = conn.createStatement()) {
            String query = "SELECT * FROM users WHERE username = '" + user + "'";
            ResultSet rs = stmt.executeQuery(query);
            while (rs.next()) {
                resp.getWriter().println("User: " + rs.getString("username"));
            }
        } catch (SQLException e) {
            resp.getWriter().println("DB error");
        }

        // --- VULN: Command injection via Runtime.exec with user input ---
        String host = req.getParameter("host"); // user-controlled
        try {
            // Dangerous: passes user input into shell command
            Process p = Runtime.getRuntime().exec("ping -c 1 " + host);
            BufferedReader br = new BufferedReader(new InputStreamReader(p.getInputStream()));
            String line;
            while ((line = br.readLine()) != null) {
                resp.getWriter().println(line);
            }
        } catch (Exception e) {
            resp.getWriter().println("Exec failed");
        }
    }
}
